src/components/ui/dialog-modal.tsx

```diff
@@ -0,0 +1,225 @@
+       */
+      "data-[state=open]:animate-contentShow size-[calc(-2rem_+_100%)] fixed left-1/2 top-1/2 grid -translate-x-1/2 -translate-y-1/2 gap-4 rounded-md bg-background p-8 shadow-[hsl(206_22%_7%_/_35%)_0px_10px_38px_-10px,_hsl(206_22%_7%_/_20%)_0px_10px_20px_-15px] focus:outline-none z-modal  overflow-auto",
+    overlay:
+      "data-[state=open]:animate-overlayShow fixed inset-0 bg-black/70 z-overlay",
+    header: "relative pe-12",
+    title: "text-2xl",
+    footer: "pt-8",
+    close: "text-md size-8 z-10 absolute end-0 top-0",
+  },
+  variants: {
+    size: {
+      md: {
+        content: "max-w-xl",
+      },
+      xl: {
+        content: "max-w-[1004px]",
+      },
+    },
+    isSimulator: {
+      true: {
+        close: "static ms-auto",
+        header: "pe-0",
+      },
+    },
+  },
+  defaultVariants: {
+    size: "md",
+  },
+})
+
+type DialogVariants = VariantProps<typeof dialogVariant>
+
+const DialogStylesContext = React.createContext(dialogVariant())
+
+const useDialogStyles = () => React.useContext(DialogStylesContext)
+
+type DialogProps = DialogPrimitive.DialogProps & DialogVariants
+
+const Dialog = ({ size, isSimulator, ...props }: DialogProps) => {
+  const styles = dialogVariant({ size, isSimulator })
+  return (
+    <DialogStylesContext.Provider value={styles}>
+      <DialogPrimitive.Root {...props} />
+    </DialogStylesContext.Provider>
+  )
+}
+
+const DialogTrigger = DialogPrimitive.Trigger
+
+const DialogPortal = DialogPrimitive.Portal
+
+const DialogClose = DialogPrimitive.Close
+
+const DialogOverlay = React.forwardRef<
+  React.ElementRef<typeof DialogPrimitive.Overlay>,
+  React.ComponentPropsWithoutRef<typeof DialogPrimitive.Overlay>
+>(({ className, ...props }, ref) => {
+  const { overlay } = useDialogStyles()
+  return (
+    <DialogPrimitive.Overlay
+      ref={ref}
+      className={cn(overlay(), className)}
+      {...props}
+    />
+  )
+})
+DialogOverlay.displayName = DialogPrimitive.Overlay.displayName
+
+type DialogContentProps = React.ComponentPropsWithoutRef<
+  typeof DialogPrimitive.Content
+>
+
+const DialogContent = React.forwardRef<
+  React.ElementRef<typeof DialogPrimitive.Content>,
+  DialogContentProps
+>(({ className, children, ...props }, ref) => {
+  const { content } = useDialogStyles()
+  return (
+    <DialogPortal>
+      <DialogOverlay />
+      <DialogPrimitive.Content
+        ref={ref}
+        className={cn(content(), className)}
+        {...props}
+      >
+        {children}
+      </DialogPrimitive.Content>
+    </DialogPortal>
+  )
+})
+DialogContent.displayName = DialogPrimitive.Content.displayName
+
+const DialogHeader = ({
+  className,
+  children,
+  ...props
+}: React.HTMLAttributes<HTMLDivElement>) => {
+  const { header, close } = useDialogStyles()
+  return (
+    <div className={cn(header(), className)} {...props}>
+      {children}
+      <Center className={close()} asChild>
+        <DialogPrimitive.Close>
+          <MdClose size="20" />
```

Comment:
This close button is different from prod, because the prod version comes from Chakra, which has it's own close icon, and different than `MdClose`. Therefore, I chose to use react-icons `size` prop to set a different value to match prod.

Reply:
Looks good!